repo/godep.go

```diff
@@ -0,0 +1,55 @@
+package repo
+
+import (
+	"encoding/json"
+	"io/ioutil"
+
+	"github.com/bazelbuild/bazel-gazelle/label"
+)
+
+type goDepLockFile struct {
+	ImportPath   string
+	GoVersion    string
+	GodepVersion string
+	Packages     []string
+	Deps         []goDepProject
+}
+
+type goDepProject struct {
+	ImportPath string
+	Rev        string
+}
+
+func importRepoRulesGoDep(filename string) ([]Repo, error) {
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	file := goDepLockFile{}
+
+	if err := json.Unmarshal(data, &file); err != nil {
+		return nil, err
+	}
+
+	var repos []Repo
+	cache := NewRemoteCache(repos)
```

Comment:
The reason I'm using a RemoteCache here is that GoDep considers each package a separate repository - I.e. you can have Godep.json entries for the same repo several times, like:

```
    {
      "ImportPath": "github.com/go-redis/redis",
      "Comment": "v6.5.6-2-g7a034e1",
      "Rev": "7a034e1609674d5eb847c3885e5058c54e79a1df"
    },
    {
      "ImportPath": "github.com/go-redis/redis/internal",
      "Comment": "v6.5.6-2-g7a034e1",
      "Rev": "7a034e1609674d5eb847c3885e5058c54e79a1df"
    },
    {
      "ImportPath": "github.com/go-redis/redis/internal/consistenthash",
      "Comment": "v6.5.6-2-g7a034e1",
      "Rev": "7a034e1609674d5eb847c3885e5058c54e79a1df"
    },
```

This necessitates using a RemoteCache to find the common root and deduplicate (I'm not sure what would happen if you had different Rev's specified; I suspect it would not work so it's okay to ignore the possibility)

Reply:
Hmm, interesting. Yeah, I don't think we can support multiple copies of a repo at different versions, though it would be better to print an error instead of ignoring it.

So `RemoteCache` will be necessary. As mentioned above, please create a `RemoteCache` in update-repos.go and pass it in via `ImportRepoRules`. It's important to create the `RemoteCache` there, since it can incorporate information from WORKSPACE. This will require changing the type signature of other import functions, too, but I think it's the right thing to do. There has been interest in doing this in other changes.

One last thing: since this is going out to the network for potentially a large number of packages, please call `RemoteCache` methods in parallel. There's an example of doing this in update-repos.go (`updateImportPaths`).